db/migrate/20150612135546_create_host_status.rb

```diff
@@ -11,7 +11,29 @@ def up
     add_column :hosts, :global_status, :integer, :default => 0, :null => false
 
     Host.includes(:host_statuses, :last_report_object).find_each do |host|
-      host.refresh_statuses
+      begin
+        host.importing_facts = true
+        HostStatus.status_registry.each do |status_class|
+          status = host.get_status(status_class)
+          begin
+            status.refresh! if status.relevant?
+          rescue => e
+            # if the status is not ready to be saved because of missing migration
+            # we skip it and let it refresh by itself
+            logger.warn "skipping sub-status #{status.class} refresh, it's not ready - #{e.message}"
```

Comment:
it might generate log for every "host x status" but if you want I could print one warning if there's any issue, otoh it's just warning, statuses will be updated later by normal updates